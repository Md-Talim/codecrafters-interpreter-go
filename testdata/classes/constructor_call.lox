// Test Case 1
class Default {
  init() {
    this.x = "bar";
    this.y = 91;
  }
}
print Default().x;
print Default().y;

// Test Case 2
class Robot {
  init(model, function) {
    this.model = model;
    this.function = function;
  }
}
print Robot("R2-D2", "Astromech").model;

// Test Case 3
class Counter {
  init(startValue) {
    if (startValue < 0) {
      print "startValue can't be negative";
      this.count = 0;
    } else {
      this.count = startValue;
    }
  }
}

var instance = Counter(-52);
print instance.count;
print instance.init(52).count;

// Test Case 4
class Vehicle {
  init(type) {
    this.type = type;
  }
}

class Car {
  init(make, model) {
    this.make = make;
    this.model = model;
    this.wheels = "four";
  }

  describe() {
    print this.make + " " + this.model +
    " with " + this.wheels + " wheels";
  }
}

var vehicle = Vehicle("Generic");
print "Generic " + vehicle.type;

var myCar = Car("Toyota", "Corolla");
myCar.describe();