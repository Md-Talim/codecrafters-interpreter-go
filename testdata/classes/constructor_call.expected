bar
91
R2-D2
startValue can't be negative
0
52
Generic Generic
Toyota Corolla with four wheels